workshop.pl

```diff
@@ -455,55 +465,51 @@ sub arg_handler {
     exit(get_exit_code('no_userenv'));
 }
 
-my $json_v;
-
-my $userenv_json;
-
 my $command;
 my $command_output;
-my $rc;
-
 my $dirname = dirname(__FILE__);
 my $schema_location = $dirname . "/schema.json";
-if (! -e $schema_location) {
-    logger('error', "Failed to locate schema.json (assumed location is '$schema_location')\n");
-    exit(get_exit_code('schema_not_found'));
-} else {
-    logger('info', "Using '$schema_location' for JSON input file schema validation\n");
-
-    $json_v = JSON::Validator->new;
-    $json_v->schema($schema_location);
-}
 
+logger('info', "Using '$schema_location' for JSON input file schema validation\n");
 logger('info', "Loading userenv definition from '$args{'userenv'}'...\n");
-
 logger('info', "importing JSON...\n", 1);
-if (open(my $userenv_fh, "<", $args{'userenv'})) {
-    my $file_contents;
-    while(<$userenv_fh>) {
-        $file_contents .= $_;
-    }
-
-    $userenv_json = decode_json($file_contents);
-
-    close($userenv_fh);
 
+(my $rc, my $userenv_json) = get_json_file($args{'userenv'}, $schema_location);
+if ($rc == 0 and defined $userenv_json) {
     logger('info', "succeeded\n", 2);
 } else {
     logger('info', "failed\n", 2);
-    logger('error', "Could not open userenv file '$args{'userenv'}' for reading!\n");
-    exit(get_exit_code('failing_opening_userenv'));
-}
-
-logger('info', "validating JSON...\n", 1);
-my @json_v_errors = $json_v->validate($userenv_json);
-if (scalar(@json_v_errors) > 0) {
-    logger('info', "failed\n", 2);
-    logger('error', "Could not JSON validate the userenv file '$args{'userenv'}'!\n");
-    logger('error', Dumper(\@json_v_errors));
-    exit(get_exit_code('userenv_failed_validation'));
-} else {
-    logger('info', "succeeded\n", 2);
+    if ($rc == 1) {
+        logger('error', "Schema file undefined!\n");
+        exit(get_exit_code('no_schema'));
+    } elsif ($rc == 2) {
+        logger('error', "Schema file $schema_location not found!\n");
+        exit(get_exit_code('schema_not_found'));
+    } elsif ($rc == 3) {
+        logger('error', "Cannot open schema file: $schema_location\n");
+        exit(get_exit_code('failed_opening_schema'));
+    } elsif ($rc == 4) {
+        logger('error', "Schema $schema_location is invalid JSON!\n");
+        exit(get_exit_code('schema_invalid_json'));
+    } elsif ($rc == 5) {
+        logger('error', "Schema validation for userenv $args{'userenv'} using schema '$schema_location' failed!\n");
+        exit(get_exit_code('userenv_failed_validation'));
+    } elsif ($rc == 6) {
+        logger('error', "Userenv $args{'userenv'} is invalid JSON!\n");
+        exit(get_exit_code('userenv_invalid_json'));
+    } elsif ($rc == 7) {
+        logger('error', "Userenv file name undefined\n");
+        exit(get_exit_code('no_userenv'));
```

Comment:
this case is already covered, see:

https://github.com/perftool-incubator/workshop/blob/3e5b16b0e574e17659501176cf17e17a22c50d59/workshop.pl#L452-L456